app/src/main/java/org/alphatilesapps/alphatiles/Validator.java

```diff
@@ -0,0 +1,992 @@
+                if (entry.getValue() < 6) {
+                    warnings.add("In wordList.txt, the letter " + entry.getKey() + " is only used" +
+                            " in " + entry.getValue() + " words. It is recommended that each letter be" +
+                            " used in at least 6 words.");
+                }
+            }
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+
+        //todo is checking headers still a necessary check for gameTiles or others?.. aslo non initial letters black box error?
+        try {
+            // the below section compares the tiles in gameTiles to the words in wordlist
+            Map<String, Integer> tileUsage = new HashMap<>();
+            int longWords = 0;
+            tileUsage.put(".", 0);
+            tileUsage.put("'", 0);
+            for (String tile : langPackGoogleSheet.getTabFromName("gametiles").getCol(0)) {
+                tileUsage.put(tile, 0);
+            }
+            for (String word : langPackGoogleSheet.getTabFromName("wordlist").getCol(1)) {
+                // the below numTilesInWord method parses a word into the tiles in the tileUsage dictionary,
+                // returning the number of tiles in the provided word.
+                // it also updates the tileUsage dictionary so the dictionary counts how many times each tile appears
+                int tileCounter = numTilesInWord(word, tileUsage);
+                if (tileCounter == 0) {
+                    warnings.add("no combination of tiles can be put together to create " + word + "in wordlist");
+                }
+
+                if (tileCounter >= 10) {
+                    if (tileCounter > 15) {
+                        fatalErrors.add("the word " + word + " in wordlist takes more than 15 tiles to build");
+                    } else {
+                        longWords += 1;
+                    }
+                }
+            }
+            if (longWords > 0) {
+                warnings.add("the wordlist has " + longWords + " long words (10 to 15 game tiles);" +
+                        " shorter words are preferable in an early literacy game. Consider removing longer words ");
+            }
+            for (Map.Entry<String, Integer> tile : tileUsage.entrySet()) {
+                if (tile.getValue() < 3) {
+                    warnings.add("the tile " + tile.getKey() + " in gametiles only appears in words " + tile.getValue()
+                            + " times. It is recommended that each letter be used in at least 3 words");
+                }
+            }
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+        try {
+            //todo.. the correct uppercase value?
+            Tab gameTiles = langPackGoogleSheet.getTabFromName("gametiles");
+            // below helper methods adds warnings for any duplicates in the provided column
```

Comment:
```suggestion
            // Add warnings for any duplicates in the provided column
```
I recommend imperative mood for quick comments